src/puppetlabs/dujour/version_check.clj

```diff
@@ -88,10 +88,15 @@
                          (merge version-data)
                          json/generate-string)
         url update-server-url
-        {:keys [status body] :as resp} (client/post url
-                                         {:headers {"Accept" "application/json"}
-                                          :body request-body
-                                          :as :text})]
+        {:keys [status body] :as resp} (try
+                                         (client/post url
+                                                     {:headers {"Accept" "application/json"}
+                                                      :body request-body
+                                                      :as :text})
+                                         (catch java.io.IOException e
+                                           (throw+ {:type ::connection-error
+                                                    :message (.getMessage e)
+                                                    :exception e})))]
```

Comment:
Yes please.  We've been slowly changing all of these over to the other format so I'd really like to avoid introducing new instances that don't comply.  The next release of kitchensink will switch over to the `:kind` format, and at that point I'm not aware of any places left where we're not adhering to that.